addons/account/models/account_move.py

```diff
@@ -4176,6 +4176,8 @@ def _get_fields_onchange_subtotal_model(self, price_subtotal, move_type, currenc
         :param date:            The move's date.
         :return:                A dictionary containing 'debit', 'credit', 'amount_currency'.
         '''
+        if self.env.context.get('use_invoice_date_for_conversion', False):
+            date = self.move_id.invoice_date or date
```

Comment:
Instead of that, replace in all calls to `_get_fields_onchange_subtotal_model` the date by `move.invoice_date or move.date`

Reply:
Hey Laurent !
umm, CHKL said he wants to use the invoice date only when it's set automatically (on post) and there is no solution for that (without breaking stable) except using context, but correct me if I am wrong please !